src/main/groovy/io/seqera/wave/service/counter/impl/LocalCounterProvider.groovy

```diff
@@ -49,10 +50,11 @@ class LocalCounterProvider implements CounterProvider {
 
     @Override
     Map<String, Long> getAllMatchingEntries(String key, String pattern) {
-        def keyStore = store.get(key)
+        Pattern compiledPattern = Pattern.compile(pattern.replace('*', '.*'))
```

Comment:
usually makes sense to declares as a static field to avoid to recompile each time

Reply:
in this case it is using method parameter inside, that's why its not static